src/Servers/HttpSys/test/FunctionalTests/Http3Tests.cs

```diff
@@ -0,0 +1,123 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.Net.Http.Headers;
+using Xunit;
+
+// Not tested here: Http.Sys supports sending an altsvc HTTP/2 frame if you enable the following reg key.
+// reg add "HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\services\HTTP\Parameters" /v EnableAltSvc /t REG_DWORD /d 1 /f
+// However, this only works with certificate bindings that specify a name. We test with the IP based bindings created by IIS Express.
+// I don't know if the client supports the HTTP/2 altsvc frame.
+namespace Microsoft.AspNetCore.Server.HttpSys
+{
+    [MsQuicSupported] // Required by HttpClient
+    [Http3Supported]
+    public class Http3Tests
```

Comment:
Is there any need to test features like reset and trailing headers still work with HTTP/3?

Reply:
Added. I ended up adding one special case in the code for abort codes.